src/shillelagh/backends/apsw/vt.py

```diff
@@ -191,6 +191,10 @@ def get_all_bounds(
         column_type = columns[column_name]
 
         # convert constraint to native Python type, then to DB specific type
+        if isinstance(constraint, set):
+            # A WHERE x IN (list) clause sends in constraintargs as a set when N>1:
+            # {item1,...,itemN}. We serialize this set to a string '["item1",...,"itemN"]'.
+            constraint = json.dumps(list(constraint))
```

Comment:
ok, but why do you have to serialize this at all here?

Reply:
As far as I can tell only strings pass through apsw.

Reply:
I would suggest writing a test sooner than later as it will help with this.

Reply:
No, that won't work - the set is a choice of values. This would make it an 'And' operation.

Reply:
sorry, what won'y work? `EQ` is what sqlite / apsw pass in (I believe reading the docs). I am asying you need to pass along tothe adapter that this is an `IN`

Reply:
If the constraint comes in as a set: {'a', 'b'}.
then constraint = [type_map[column_type.type]().parse(c) for c in constraint]
converts it to ['a', 'b'] for a TEXT column type.
then value = [column_type.format(c) for c in constraint]
converts it to ['a', 'b']
You can't add a list to the EQ operator - it has to be a string . That's why I was serializing the list.
value = jsonj.dumps([column_type.format(c) for c in constraint]) does work.

Reply:
What do you mean you can't "add"  a list to the EQ operator?

Reply:
Yes that looks like putting a list into a set but what line is causing that.

Reply:
`all_bounds[column_name].add((operator, value))` which `value` is a list so `(operator, value)` is unhashable

Reply:
So change this line:
```
value = [column_type.format(c) for c in constraint]
```
To
```
value = tuple(column_type.format(c) for c in constraint)
```

Reply:
Or use frozenset: https://docs.python.org/3/library/stdtypes.html#frozenset

Reply:
Yes that should work, then we got adding new type of operator left, waiting for the update and merge ^^

Reply:
Went with tuple() and added the Operator.IN